### PR TITLE
feat: remove from project

### LINKED ISSUE #721: feat: $removeFromProject built-in
**Is your feature request related to a problem? Please describe.**
It would be useful to remove PR from Github Project when PR is closed without merging, to undo adding PR to GitHub Project that could have been executed when PR was created.

**Describe the solution you'd like**
New built-in could be created, smth like
`$removeFromProject("my project")`
to revert for example
`$addToProject("my project", "Review needed")`

**Describe alternatives you've considered**
Alternatively we could 
a) use the same GitHub Project status for both PRs that are merged and closed without merging. Concidering small amount of PRs that are closed without merging it isn't too annoying, but having closed non-merged PRs on the board has no value.
b) create separate status for GitHub Project for PRs closed without merging, but it wouldn't add much value for the project overview.

**Additional context**
@ferreiratiago suggested me to submit this feature request when i asked about it from [chat](https://discord.com/channels/935103393368801351/935120672810750022/1077863613047525377)

### PR DESCRIPTION
**Pull request by zolamk**
## Description
Adds a new built-in `$removeFromProject` that allows removing an issue / PR from a project with a specified name.

## Related issue

Closes #721

## Type of change

New feature (non-breaking change which adds functionality)

## How was this tested?
Unit & Manual tests.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have ran `task check -f` and have no issues

## Code review and merge strategy (ship/show/ask) 

 
 
- [x] Ask: this pull request requires a code review before merge

### CODE REVIEW

**Review by shay2025 — APPROVED** [review]

File: codehost/github/projects.go [hunk 1]
```diff
@@ -86,6 +87,43 @@ type FieldDetails struct {
...
+					Project struct {
+						ID string
+					}
+					ID string
+				}
+			} `graphql:"projectItems(first: 100, after: $afterCursor)"`
+		} `graphql:"pullRequest(number: $number)"`
+	} `graphql:"repository(owner: $owner, name: $name)"`
+}
+type GetIssueProjectItemsQuery struct {
```
Comment (ferreiratiago) [inline codehost/github/projects.go:111]: Can we add an empty line before this?

File: codehost/github/projects.go [hunk 2]
```diff
@@ -166,3 +204,77 @@ func (c *GithubClient) GetProjectFieldsByProjectNumber(ctx context.Context, owne
 
 	return fields, nil
 }
+
+func (c *GithubClient) GetPullRequestProjectV2ItemIDByNumber(ctx context.Context, owner, repo, projectID string, number int) (string, error) {
```
Comment (ferreiratiago) [inline codehost/github/projects.go:208]: Can we please add a comment before explaining what this function does?

```go
// GetPullRequestProjectV2ItemIDByNumber returns the project item ID for a pull request.
// Each item on a project board is represented by a project item.
```

Also, I'm not sure if the `byNumber` here helps. I would suggest to rename it:
```suggestion
func (c *GithubClient) GetPullRequestProjectV2ItemID(ctx context.Context, owner, repo, projectID string, prNum int) (string, error) {
```

WDYT?

File: codehost/github/projects.go [hunk 3]
```diff
@@ -166,3 +204,77 @@ func (c *GithubClient) GetProjectFieldsByProjectNumber(ctx context.Context, owne
...
+			break
+		}
+
+		varGQLGetPullRequestProjectItemsQuery["afterCursor"] = githubv4.String(getPullRequestProjectItemsQuery.Repository.PullRequest.ProjectItems.PageInfo.EndCursor)
+	}
+
+	return "", ErrProjectItemNotFound
+}
+
+func (c *GithubClient) GetIssueProjectV2ItemIDByNumber(ctx context.Context, owner, repo, projectID string, number int) (string, error) {
```
Comment (ferreiratiago) [inline codehost/github/projects.go:240]: The same here.

```go
// GetIssueProjectV2ItemIDByNumber returns the project item ID for an issue.
// Each item on a project board is represented by a project item.
```

```suggestion
func (c *GithubClient) GetIssueProjectV2ItemID(ctx context.Context, owner, repo, projectID string, issueNum int) (string, error) {
```

**Review by ferreiratiago — COMMENTED** [review]
Minor suggestion. Please let me know WDYT.

**Review by ferreiratiago — APPROVED** [review]

**Pull request merged**